connectors/src/connectors/github/lib/utils.ts

```diff
@@ -8,34 +8,43 @@ export const GITHUB_CONTENT_NODE_TYPES = [
   "REPO_CODE_DIR",
   "REPO_CODE_FILE",
 ] as const;
+
 export type GithubContentNodeType = (typeof GITHUB_CONTENT_NODE_TYPES)[number];
 
+export function isGithubCodeDirId(internalId: string): boolean {
+  return /^github-code-\d+-dir-[a-f0-9]+$/.test(internalId);
+}
+
+export function isGithubCodeFileId(internalId: string): boolean {
+  return /^github-code-\d+-file-[a-f0-9]+$/.test(internalId);
+}
+
 /**
  * Gets the type of the Github content node from its internal id.
  */
 export function matchGithubNodeIdType(internalId: string): {
   type: GithubContentNodeType;
   repoId: number;
 } {
-  // Full repo is selected, format = "12345678"
-  if (/^\d+$/.test(internalId)) {
+  // Full repo is selected, format = "github-repository-12345678"
+  if (/^github-repository-\d+$/.test(internalId)) {
```

Comment:
1. are you confident with the regexes?
2. are those used elsewhere apart from migrations? if yes, we should have isGithubXXX just lice for code dir / file

Reply:
I borrowed the regex from existing code in the connector, I'm pretty confident anyway

Reply:
They are used in connectors but nowhere in front